src/app/vet/(logged-in)/schedule/_components/appointment-scheduler.tsx

```diff
@@ -18,9 +18,20 @@ export function AppointmentScheduler(props: { dogs: AvailableDog[] }) {
   const { dogs, selectedDogId, outcomes } = schedulerState;
 
   function selectDog(dogId: string) {
+    if (selectedDogId === dogId) {
+      closeCallCard();
+    } else {
+      setSchedulerState({
+        ...schedulerState,
+        selectedDogId: dogId,
+      });
+    }
+  }
+
+  function closeCallCard() {
```

Comment:
When do we use this?

Reply:
It was called from the selectDog function.

Both have been refactored into toggleDogSelection in e02c1e5